uadi.h

```diff
@@ -0,0 +1,63 @@
+#ifndef UADI_H
+#define UADI_H
+
+#include <stddef.h>
+
+#ifdef _WIN32
+#define DLL_EXPORT __declspec(dllexport)
+#else
+#define DLL_EXPORT
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void* uadi_lib_handle;
+typedef void* uadi_device_handle;
+
+typedef unsigned char* uadi_chunk_ptr;
+
+typedef struct uadi_receive_struct{
+    uadi_chunk_ptr infopack_ptr;
+    uadi_chunk_ptr datapack_ptr;
+};
+
+// Error codes
+#define UADI_SUCCESS 0
+#define UADI_ERROR -1
+#define UADI_BUFFER_TOO_SMALL -2
+#define UADI_INVALID_HANDLE -3
+#define UADI_NO_DATA -4
+#define UADI_OUT_OF_CHUNKS -5
+#define UADI_INTERNAL_ERROR -255
+
+typedef int uadi_status;
+
+typedef void(uadi_error_callback*)(uadi_status, void*);
```

Comment:
Shouldn't it be like `typedef void(*uadi_error_callback)(uadi_status, void*);` to make `uadi_error_callback` a type?

Reply:
second that